pkg/requestreply/cesql.go

```diff
@@ -27,45 +30,59 @@ import (
 
 	secretinformer "knative.dev/pkg/injection/clients/namespacedkube/informers/core/v1/secret"
 	"knative.dev/pkg/logging"
+	"knative.dev/pkg/system"
 )
 
 var once = sync.Once{}
 
 // RegisterCESQLVerifyCorrelationIdFilter registers a new function with the CESQL runtime to verify correlation ids
-// The function signature is KN_VERIFY_CORRELATION_ID(correlation_id, namespace, secret_names...)
+// The function signature is KN_VERIFY_CORRELATION_ID(correlation_id, name, namespace, secret_name, pod_idx, replica_count)
 func RegisterCESQLVerifyCorrelationIdFilter(ctx context.Context) error {
 	secretInformer := secretinformer.Get(ctx)
 	logger := logging.FromContext(ctx)
 
 	var verifyCorrelationId = cefn.NewFunction(
 		"KN_VERIFY_CORRELATION_ID",
-		[]cesql.Type{cesql.StringType, cesql.StringType},
-		cesql.TypePtr(cesql.StringType),
+		[]cesql.Type{cesql.StringType, cesql.StringType, cesql.StringType, cesql.StringType, cesql.IntegerType, cesql.IntegerType},
+		nil,
 		cesql.BooleanType,
 		func(e cloudevents.Event, args []interface{}) (interface{}, error) {
-			replyId, namespace := args[0].(string), args[1].(string)
-			secretNames := args[2:]
-
-			for _, secretName := range secretNames {
-				secret, err := secretInformer.Lister().Secrets(namespace).Get(secretName.(string))
-				if err != nil {
-					logger.Warnf("failed to get secret %s in namespace %s: %s", secretName.(string), namespace, err.Error())
-					continue
-				}
+			replyId, name, namespace, secretName, podIdx, replicaCount := args[0].(string), args[1].(string), args[2].(string), args[3].(string), args[4].(int32), args[5].(int32)
```

Comment:
The comment above says `correlationId`. In your gist/example I saw both in the response: ce-replyid and ce-correlationid.

Reply:
Good point - it is worth clarifying this. The idea is that users will set the `correlationid` for all cloudevents associated with a request, and only set the `replyid` on the final event for the request (the one that should be treated as the "reply")

I think I was calling it `correlationid` here since technically this CESQL is generic, and at runtime this variable has the _value_ of the extension attribute. Happy to align to either all correlationid or all replyid